ffc/codegeneration/jit.py

```diff
@@ -195,27 +200,35 @@ def _compile_objects(decl, ufl_objects, object_names, module_name, parameters):
     ffibuilder = cffi.FFI()
     ffibuilder.set_source(module_name, code_body, include_dirs=[ffc.codegeneration.get_include_path()],
                           extra_compile_args=['-g0'])  # turn off -g
-
     ffibuilder.cdef(decl)
 
     c_filename = compile_dir.joinpath(module_name + ".c")
     ready_name = c_filename.with_suffix(".c.cached")
 
-    # Ensure path is set for module and ensure cache dir exists
-    sys.path.insert(0, str(compile_dir))
+    # Compile (ensuring that compile dir exists)
     compile_dir.mkdir(exist_ok=True)
-
-    # Compile
     ffibuilder.compile(tmpdir=compile_dir, verbose=False)
 
     # Create a "status ready" file. If this fails, it is an error,
     # because it should not exist yet.
     fd = open(ready_name, "x")
     fd.close()
 
-    # Build list of compiled objects
-    compiled_module = importlib.import_module(module_name)
-    sys.path.remove(str(compile_dir))
+    # Create module finder that searches the compile path
+    finder = importlib.machinery.FileFinder(
+        str(compile_dir), (importlib.machinery.ExtensionFileLoader, importlib.machinery.EXTENSION_SUFFIXES))
+
+    # Find module. Clear search cache to be sure dynamically created
+    # (new) modules are found
+    finder.invalidate_caches()
+    spec = finder.find_spec(module_name)
+    if spec is None:
+        raise ModuleNotFoundError("Unable to find JIT module.")
+
+    # Load module
+    compiled_module = importlib.util.module_from_spec(spec)
+    spec.loader.exec_module(compiled_module)
+
```

Comment:
Since this is repeated code, I wonder if it would be better in a small function, e.g. `def _load_module(name, path)`

Reply:
I'd rather keep it as-is for readability and clarity. Also, I would like to revise the function `get_cached_module`.